kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/balance/view/BalanceView.java

```diff
@@ -1,120 +1,260 @@
+	private static int subTitleColor;
+
+	private static float balanceTextSize;
+	private static float subTitleTextSize;
+
+	private IBalancePresenter balancePresenter;
+
+	private TextSwitcher subTitleTextView;
+	private TextSwitcher balanceText;
+	private ImageView arrow;
+
+	private float arrowFromX = 0;
+	private float arrowToX = 0;
+
+	public BalanceView(Context context) {
+		super(context);
+		init(context, null);
+	}
+
+	public BalanceView(Context context, AttributeSet attrs) {
+		super(context, attrs);
+		init(context, attrs);
+	}
+
+	public BalanceView(Context context, AttributeSet attrs, int defStyleAttr) {
+		super(context, attrs, defStyleAttr);
+		init(context, attrs);
+	}
+
+	private void init(Context context, @Nullable AttributeSet attrs) {
+		inflate(context, R.layout.kinecosystem_balance_view, this);
+		setLayoutParams(new ConstraintLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+			ViewGroup.LayoutParams.WRAP_CONTENT));
+		int topPadding = getResources().getDimensionPixelOffset(R.dimen.kinecosystem_main_medium_margin);
+		setPadding(0, topPadding, 0, 0);
+		initColors();
+		initSizes();
+
+		TypedArray styledAttributes = context.getTheme()
+			.obtainStyledAttributes(attrs, R.styleable.KinEcosystemBalanceView, 0, 0);
+		String subtitle;
+		try {
+			subtitle = styledAttributes.getString(R.styleable.KinEcosystemBalanceView_subTitle);
+
+		} finally {
+			styledAttributes.recycle();
+		}
+
+		this.subTitleTextView = findViewById(R.id.sub_title);
+		this.balanceText = findViewById(R.id.balance_text);
+		this.arrow = findViewById(R.id.arrow);
+		addBalanceViewFactory();
+		addSubTitleViewFactory();
+		setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				balancePresenter.balanceClicked();
+			}
+		});
+		this.arrow.getViewTreeObserver().addOnGlobalLayoutListener(new OnGlobalLayoutListener() {
+			@Override
+			public void onGlobalLayout() {
+				if (arrow.getMeasuredWidth() > 0) {
+					arrowFromX = arrow.getPivotX() - getResources().getDimensionPixelOffset(R.dimen.kinecosystem_main_medium_margin);
+					arrowToX = arrowFromX + arrow.getWidth();
+					arrow.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+				}
+			}
+		});
+
+		this.subTitleTextView.setText(subtitle);
+	}
+
+	private void initColors() {
+		blueColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_bluePrimary);
+		orangeColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_orange);
+		redColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_red);
+		subTitleColor = ContextCompat.getColor(getContext(), R.color.kinecosystem_subtitle);
+	}
+
+	private void initSizes() {
+		Resources resources = getResources();
+		balanceTextSize = resources.getDimension(R.dimen.kinecosystem_title_xlarge_size);
+		subTitleTextSize = resources.getDimension(R.dimen.kinecosystem_sub_title_size);
+	}
+
+
+	private void addBalanceViewFactory() {
+		balanceText.setFactory(new ViewFactory() {
```

Comment:
What not using simpler sin text of  addView

Reply:
sin ?
But this is simple wat to have also change animation between 2 textviews.